### ISSUE TITLE
Minimizing Dimensional Emptiness 

### ISSUE THREAD

**Issue by Viuss** [Tslat/Advent-Of-Ascension#1895]
One of the biggest issues I have seen that people usually have with this mod is that it feels like every dimension is super simple, boring, and empty. While I think there is definitely content there, I kind of agree (still love this mod to death though). Granted, over the years there has definitely been a massive improvement, especially from tslat. But to be honest, it is really difficult to fix a system that is built so heavily around infinite new areas for the player. However, I think the content in the mod is really great, but it is a bit spread out. I know you are always looking to add new things, but compression might go a long way. 

With 22 dimensions, there are many concepts that, at their very basic level, feel repeated. Dimensions like Greckon, Dustopia, and the Abyss all fill this niche of "evil dark place with scary mobs you have a hard time seeing" and so going from one to another, especially since you basically visit them in tandem, feels repetitive, especially if the player isn't finding much in them. By cutting out 2 of the 3 and combining their features, not only does it solve the issues related to tedious world jumping, but it fills them with much more content that makes the player more likely to spend extra time there and pay attention to what is happening. Right now, each dimensions feels like just "find the boss items, fight the boss, maybe do some shopping, then go to the next one." That's really it, for dimensions like the ones I referenced, Mobs from Dustopia and Greckon like the Arkzyne and Hunter already have very similar aesthetics, and you wouldn't really be able to tell one was out of place if you found it in one of the other two dimensions. By taking the content from two of these and transferring them completely to the 3rd, you get a dimension that is now packed full of varying creatures, has many more types of structures (that hopefully spawn a bit more often), and feels like a real place the player might want to stay for more than just an hour or two. When I use this mod in packs, I make sure to enable as many structure gen and mob spawn configs as I can to spawn in AoA worlds because it just make it feel livelier, and it does so much. Essentially my argument is that many dimensions have features at their core that are very similar to others, and can easily be combined. I know that shifting between worlds is a big part of this mods identity, and I don't want to change that, but I've literally used this mod for several years now and even in the older days with much less content I was not able to finish it, I wasn't able to even visit 3/4 of what it had to offer, and people I tried to use it with got bored before they got further in. Even if you want to keep most of them, cutting down to 12 or 15 would be a much smoother and much needed change. I can even recommend the exact combinations
**Dustopia / Greckon** / Abyss - All based around shadowy nightmarish monsters, limited sight, eerie
**Lunalus / Mysterium** - Both very purple with many creepy, bulbous creatures. the mushrooms could easily be become parts of planets, and the water is typically unnecessary. They both give a very similar "unearthly" vibe. 
**Immortalis / Iromine** - Both feel steampunk / technological in their own way. I would love to keep the tasks system that immortalis has but iromine's mobs could very easily fit in, and their color schemes are also similar.
**Barathos / L'Borean** - Barathos could benefit from some more personality, it feels like more of a template than it's own dimension. I know it is supposed to be desert, but just looking at it just doesn't elicit that response. It would also be possible to have Barathos be a desert with L'Borean style oasis' scattered.
**Candyland / Celeve** - I'm sorry but candyland is an assault on the eyes and Celeve is a monstrosity (clowns are just the worst). I think they share a common "childish but also terrifying" concept and candy and clowns just screams for a theme park dimension.
**Haven / Lelyetia** - When playing through Haven after Lelyetia, my first thought was "This just feels palette swapped and more mountainous." Picking a single color or even combining the blue, orange, and yellow (blue is the only one that doesn't exist in both) would help, a lot of the mobs are "floaty" and could easily work in both (maybe with minor color changes).
**Deeplands / Ancient Cavern** - The way the Deeplands looks is amazing, Ancient cavern has great bosses that could easily fit in, it would be such a perfect experience.
**Gardencia / Runandor / Shryelands** - The three of these feel like they give off the "mythical magical forest" feeling, and it would be and all have mobs and buildings that could work in others. If I saw a Runicorn or Paladin in the Shryelands I wouldn't think twice. I think Gardencia is one of the most empty dimensions, it's hard to find stuff to fit the food theme, but it has plenty of great plant based mobs and structures that could easily be moved to fit in with the Shryelands forest style. (Or you could make some of them creepers and plop them in creeponia, that place has a serious lack of mobs but it's style is so unique I couldn't think of how you might combine it with something)

All in all I hope my suggestions at least get some people thinking, I myself am not a modder so I don't know the hoops you might have to jump through to implement some of these things. However it just feels that the best parts of my favorite mod are spread too thin. Even axing some dimensions so you can put more focus on the best ones would help a lot. It might also cut down on some of the file space and reduce lag on servers since they won't have to load as many dimensions. If not, I understand, I just really want this mod to be incredible and I trust the developers will do what is best for it. 

Hopefully my suggestion is not one that pops up a lot, but I really do think combining and compressing aspects of the mod will make it that much more enjoyable for newer players, especially those who don't have the time to sprint endlessly through empty spaces and kill the same 4 mobs over and over. I know there is a 1.15 and 1.16 port coming up, and I think a version switch might be the perfect time to make big changes like this.

**Comment by EatingSilencer** [comment]
The only one of these that I think would work is Abyss/Dustopia/Greckon. Some of these would not be easily mergeable due to some issues, mainly that the dimensions are different tiers on different paths. I also don't agree with some of these merges.

A dimension being on a spot in the progression tree isn't just for funsies, it affects:
1. the mobs are balanced to be harder.
2. the weapons are balanced to deal with the stronger mobs

Abyss is tier 2 while Greckon/Dustopia are tier 3, merging them would mean that the mobs of either Abyss or Greckon/Dustopia would have to be rebalanced to fit the difficulty level of the other, and the weapons rebalanced to fit as well. Assuming this is the only change that happens, this means that shyrelands gets demoted to tier 3 and all its mobs and weapons have to be rebalanced too.

I'm not sure I understand immortallis/iromine, you would be merging a fixed dungeon dimension with a normal surface dimension...how would that work? Immortallis might be mergeable with ancient cavern or labricon but that's it I think. 

Also, disagree on ancient cavern bosses being able to fit in with deeplands. Same case as immortallis/iromine, ancient cavern is an ancient temple of the gods, and the bosses are their guardians...in a rocky dimension like deeplands? No thanks. Would also mean that deeplands' position in the progression would need to be rethunk.

Barathos/L'borean is a wild stretch too, you can add oasis to barathos while leaving l'borean alone.

I know some people aren't big fans of candyland but I personally think its fine as it is, if you can look past its appearance and take the lore into consideration (that's its not a dimension made of candy, its a dimension that looks like its made of candy, e.g.: peppermint slugs are made of slug, not peppermint, and taste as such), then candyland is more like gardencia than it is celeve.

I also feel that merging dimensions would be signing scimiguy's death warrant. I don't know, maybe he has a different perspective on that.

These are my thoughts, hopefully they don't sound too off, I think that in order to do something like this, it would had to have been done pre-3.2, before the new progression was set in-stone, as this would basically be reversing the entirety of the loot update (redo mob balance, redo weapon balance, redo weapon crafting, heck, even redo mob drops). I also don't think doing this on a port update is wise, at least not now. AoA is porting to 1.15, but the decision to do so was done before 1.16 became any kind of real modding version, and once the port finishes aoa might have to port again. Putting an update that basically redos the loot update on a port would be a good way to make sure aoa never updates again.

tl:dr; I don't agree with this as it would basically undo the loot update.
